src/util/export.ts

```diff
@@ -0,0 +1,107 @@
+ * so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import { toPng } from "html-to-image";
+
+/**
+ * Export image of any html element.
+ * Uses a html-to-image.
+ *
+ * @param {HTMLElement} element of any component to save as image
+ * @param {ExportOptions} options are the different parameter for how
+   the image should be
+ * @returns {Blob} return the element image on clipboard
+ */
+export function exportElement(
+  element: HTMLElement,
+  options?: ExportOptions,
+): void {
+  _exportElement(element, options).catch((error) => {
+    console.error("Failed to export:", error);
+    if (options?.handleError) {
+      options.handleError("Failed to export", error);
+    }
+  });
+}
+
+export async function _exportElement(
+  element: HTMLElement,
+  options: ExportOptions = {},
+): Promise<void> {
+  if (element) {
+    const chartElement = element;
+    try {
+      if (options.format !== "png") {
+        const errorMessage = `Format '${options.format}' is not supported.`;
+        console.error(errorMessage);
+        if (options.handleError) {
+          options.handleError(errorMessage, new Error(errorMessage));
+        }
+        return;
+      }
+      const dataUrl = await toPng(chartElement, {
+        backgroundColor: "#ffffff",
+        canvasWidth: options.width || chartElement.clientWidth,
+        canvasHeight:
+          ((options.width || chartElement.clientWidth) *
+            chartElement.clientHeight) /
+            chartElement.clientWidth || options.height,
+      });
+      const response = await fetch(dataUrl);
+      const blob = await response.blob();
+
+      try {
+        await navigator.clipboard.write([
+          new ClipboardItem({
+            "image/png": blob,
+          }),
+        ]);
+        console.log("image written");
+        if (options.handleSuccess) {
+          options.handleSuccess("Image copied to clipboard.");
+        }
+      } catch (error) {
+        console.error("Clipboard write failed", error);
+        if (options.handleError) {
+          options.handleError("Clipboard write failed", error);
+        }
+      }
+    } catch (error) {
+      console.error("Failed to export:", error);
+      if (options.handleError) {
+        options.handleError("Failed to export", error);
+      }
+    }
```

Comment:
Do not catch. The calling function catches already and contains the error handler. Duplicated & redundant code otherwise.